runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java

```diff
@@ -265,28 +269,58 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		if (hotKeyPressed && event.getOption().equals(WALK_HERE))
 		{
 			MenuEntry[] menuEntries = client.getMenuEntries();
-			menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 1);
+			if (!config.clearLocalMarkers())
+			{
+				menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 1);
+
+				//Create 1 new menu entries (one for marking)
+				MenuEntry menuEntry = menuEntries[menuEntries.length - 1] = new MenuEntry();
+
+				//Set options for marking
+				menuEntry.setOption(MARK);
+				menuEntry.setTarget(event.getTarget());
+				menuEntry.setType(MenuAction.CANCEL.getId());
 
-			MenuEntry menuEntry = menuEntries[menuEntries.length - 1] = new MenuEntry();
+				//Set the new menu
+				client.setMenuEntries(menuEntries);
+			}
+			else
+			{
+				menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 2);
+
+				//Create 2 new menu entries (one for marking, one for clearing)
+				MenuEntry clearLocalMenuEntry = menuEntries[menuEntries.length - 2] = new MenuEntry();
+				MenuEntry menuEntry = menuEntries[menuEntries.length - 1] = new MenuEntry();
 
-			menuEntry.setOption(MARK);
-			menuEntry.setTarget(event.getTarget());
-			menuEntry.setType(MenuAction.CANCEL.getId());
+				//Set options for clearing
+				clearLocalMenuEntry.setOption(CLEAR_LOCAL);
+				clearLocalMenuEntry.setTarget(event.getTarget());
+				clearLocalMenuEntry.setType(MenuAction.CANCEL.getId());
 
-			client.setMenuEntries(menuEntries);
+				//Set options for marking
+				menuEntry.setOption(MARK);
+				menuEntry.setTarget(event.getTarget());
+				menuEntry.setType(MenuAction.CANCEL.getId());
+
+				//Set the new menu
+				client.setMenuEntries(menuEntries);
+			}
 		}
 	}
 
 	@Subscribe
 	public void onMenuOptionClicked(MenuOptionClicked event)
 	{
-		if (!event.getMenuOption().equals(MARK))
+		if (event.getMenuOption().equals(MARK))
 		{
-			return;
+			Tile target = client.getSelectedSceneTile();
+			markTile(target.getLocalLocation());
+		}
+		else if (event.getMenuOption().equals(CLEAR_LOCAL))
+		{
+			log.debug("Pressed clear marks");
```

Comment:
The mark option does not log any debug information, we should either remove this or ensure logging is added for both to keep it consistent.

Reply:
I'm in favor of removing this.